python/pyarrow/_csv.pyx

```diff
@@ -1433,6 +1437,18 @@ cdef class WriteOptions(_Weakrefable):
     def quoting_style(self, value):
         deref(self.options).quoting_style = unwrap_quoting_style(value)
 
+    @property
+    def quoting_header(self):
+        """
+        Same as quoting_style, but for header column names.
+        Note : both "needed" and "all_valid" have the same effect of quoting all column names.
```

Comment:
I would add this note to the `WriteOptions` docstrings above also so that it is visible in the parameters section of  the API docs.

Reply:
Thanks for the suggestion @AlenkaF. I have updated the PR with this change.